.github/workflows/pr_bot.yml

```
@@ -0,0 +1,97 @@
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: "Workflow label bot"
+on:
+  pull_request_target:
+    types:
+      - opened
+      - converted_to_draft
+      - ready_for_review
+      - synchronize
+  workflow_run:
+    workflows: ["Label when reviewed"]
+    types: ['completed']
+
+permissions:
+  contents: read
+  pull-requests: write
+  issues: write
+
+jobs:
+  pr-workflow-bot-job:
+    name: "PR Workflow bot"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Arrow
+        uses: actions/checkout@v3
+        with:
+          path: arrow
+          repository: apache/arrow
+          ref: main
+          persist-credentials: false
+          # fetch the tags for version number generation
+          fetch-depth: 0
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+      - name: Install Archery and Crossbow dependencies
+        run: pip install -e arrow/dev/archery[bot]
+      - name: 'Download PR review payload'
+        id: 'download'
+        if: github.event_name == 'workflow_run'
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const run_id = "${{ github.event.workflow_run.id }}";
+            let artifacts = await github.rest.actions.listWorkflowRunArtifacts({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: run_id,
+            });
+            let pr_review_artifact = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "pr_review_payload"
+            })[0];
+            let pr_review_download = await github.rest.actions.downloadArtifact({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              artifact_id: pr_review_artifact.id,
+              archive_format: 'zip',
+            });
+            var fs = require('fs');
+            fs.writeFileSync('${{github.workspace}}/pr_review.zip', Buffer.from(pr_review_download.data));
+      - name: Extract artifact
+        id: extract
+        if: github.event_name == 'workflow_run'
+        run: |
+          unzip pr_review.zip
+          echo "pr_review_path=$(pwd)/event.json" >> $GITHUB_OUTPUT
+      - name: Handle PR workflow event
+        env:
+          ARROW_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          if ${{ github.event_name == 'workflow_run' }}; then
```

How about using normal shell condition instead of GitHub Actions expression? `if true; then`/`if false; then` work as expected but it's not straightforward. They work because there are `/usr/bin/true` and `/usr/bin/false`.

```suggestion
          if [ "${GITHUB_EVENT_NAME}" = "workflow_run" ]; then
```